### ISSUE TITLE
Ltest-mem-validate fails on ubuntu20.04_x86

### ISSUE THREAD

**Issue by bregma** [libunwind/libunwind#391]
The `make check` test Ltest-mem-validate fails on ubuntu20.04_x86 (32-bit) in github actions.

```
2022-07-27T13:36:09.6832073Z FAIL: Ltest-mem-validate
2022-07-27T13:36:09.6832274Z ========================
2022-07-27T13:36:09.6832433Z 
2022-07-27T13:36:09.6832620Z FAIL Ltest-mem-validate (exit status: 255)
```

**Comment by bregma** [comment]
Fixed by #536

**Comment by lemul** [comment]
On FreeBSD this test sporadically fails due to it incorrectlly protect an unrelated region above process stack area, where some of the rtld data lives.
I have pushed fixes for this test (https://github.com/libunwind/libunwind/pull/571).
Fix is simple, it protects stack area which holds by consume_and_run() calls.
